common/lib/xmodule/xmodule/modulestore/tests/factories.py

```diff
@@ -324,6 +325,7 @@ def parent_location(self):  # lint-amnesty, pylint: disable=missing-function-doc
         return parent.location
 
     @classmethod
+    @strip_key
```

Comment:
This here is the biggest difference in this PR - this makes all the returned blocks from ItemFactory not have versioning attached to their location keys, which confused a lot of users in tests. This behavior matches store.get_item() and is more how people think about keys normally.

Reply:
I think the new behavior is the "obvious" one that wouldn't need an explanation (and if someone sees this method, they'll see the @split_key decorator and can follow into that to see what it does).

But chopping off the versioning is a "standard" thing when loading from the modulestore. It was weird that the factory constructors didn't do that - and was just causing weird issues in the tests where keys didn't look like code expected.